miniapps/nurbs/nurbs_curveint.cpp

```diff
@@ -0,0 +1,238 @@
+                  "Order of the NURBSPatch");
+   args.AddOption(&ifbspline, "-uw", "--unit-weight", "-nw",
+                  "--non-unit-weight",
+                  "Use a weight of one only for the NURBS-patches (default) or not: B-splines");
+
+   // Parse and print commandline options
+   args.Parse();
+   if (!args.Good())
+   {
+      args.PrintUsage(cout);
+      return 1;
+   }
+   args.PrintOptions(cout);
+
+   if (order < 2 && ifbspline == false)
+   {
+      mfem_error("For a non unity weight, the order should be at least 2.");
+   }
+
+   KnotVector *kv_o1 = UniformKnotVector(1, 2);
+   KnotVector *kv    = UniformKnotVector(order, ncp);
+
+   // 1. Create a box shaped NURBS patch
+   NURBSPatch patch(kv_o1, kv_o1, 3);
+
+   // Set weights
+   for (int j = 0; j < 2; j++)
+      for (int i = 0; i < 2; i++)
+      {
+         patch(i,j,2) = 1.0;
+      }
+
+   // Define patch corners which are box corners
+   patch(0,0,0) = -0.5*l;
+   patch(0,0,1) = -0.5*l;
+
+   patch(1,0,0) = 0.5*l;
+   patch(1,0,1) = -0.5*l;
+
+
+   patch(0,1,0) = -0.5*l;
+   patch(0,1,1) = 0.5*l;
+
+   patch(1,1,0) = 0.5*l;
+   patch(1,1,1) = 0.5*l;
+
+   // 2. Interpolation process
+   Array<Vector*> xy(2);
+   xy[0] = new Vector();
+   xy[1] = new Vector();
+   Vector xi_args, u_args;
+   Array<int> i_args;
+   xy[0]->SetSize(ncp); xy[1]->SetSize(ncp);
+
+   // Refine direction which has fitting
+   patch.DegreeElevate(0, order-kv_o1->GetOrder());
+   if (!ifbspline)
+   {
+      patch(1,0,2) = sqrt(2)/2;
+   }
+   patch.KnotInsert(0, *kv);
+
+   if (ifbspline)
+   {
+      // We locate the controlpoints at the location of the
+      // maxima of the knotvectors. This works very well
+      // for patches with the weight being one.
+      kv->FindMaxima(i_args,xi_args, u_args);
+
+      for (int i = 0; i < ncp; i++)
+      {
+         (*xy[0])[i] = u_args[i]*l;
+         (*xy[1])[i] = a * sin((*xy[0])[i]/l*2*3.1415);
+      }
+
+      kv->FindInterpolant(xy);
+
+      // Apply interpolation to patch
+      for (int i = 0; i < ncp; i++)
+      {
+         patch(i,0,0) = (*xy[0])[i]*patch(i,0,2)-0.5*l;
+         patch(i,0,1) = (*xy[1])[i]*patch(i,0,2)-0.5*l;
+      }
+   }
+
+   else
+   {
+      // We use a uniform spacing of the controlpoints. This
+      // works better for patches with not all weights being one
+      // than using the locations of the maxima.
+      for (int i = 0; i < ncp; i++)
+      {
+         (*xy[0])[i] = -0.5+(double) i * l/(ncp-1);
+         (*xy[1])[i] = -a * sin((*xy[0])[i]/l*2*3.1415)-0.5*l;
+      }
+
+      kv->FindInterpolant(xy);
+
+      // Apply interpolation to patch
+      for (int i = 0; i < ncp; i++)
+      {
+         patch(i,0,0) = (*xy[0])[i]*patch(i,0,2);
+         patch(i,0,1) = (*xy[1])[i]*patch(i,0,2);
+      }
+   }
+
+   // Refinement in curve interpolation direction
+   patch.DegreeElevate(1, order-kv_o1->GetOrder());
+   patch.KnotInsert(1, *kv);
+
+   // 3. Open and write mesh output file
+   string mesh_file;
```

Comment:
```suggestion
   string mesh_file("sin-fit.mesh");
```

Reply:
Resolved in [e31bfca](https://github.com/mfem/mfem/pull/3232/commits/e31bfca43bf510dfc9e0cd51398ea06efa9707c6)